src/mpid/ch4/shm/ipc/src/ipc_send.h

```diff
@@ -33,7 +33,10 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPC_mpi_isend(const void *buf, MPI_Aint count
 
     if (rank != comm->rank && attr.ipc_type != MPIDI_IPCI_TYPE__NONE &&
         data_sz >= attr.threshold.send_lmt_sz) {
-        if (dt_contig) {
+        MPIR_Datatype *dtp;
+        MPIR_Datatype_get_ptr(datatype, dtp);
+
+        if (dt_contig && dtp->true_lb == 0) {
             mpi_errno = MPIDI_IPCI_send_contig_lmt(buf, count, datatype, data_sz, rank, tag, comm,
```

Comment:
I do not see why we cannot support `true_lb>0` as long as it is contiguous. Can we give `sbuf+true_lb` to the receiver (i.e., using `sbuf+true_lb` to get the memory handle) ? I think this was how xpmem originally handled it. But looks like the code was lost when we refactored it to memory handle.

Reply:
Can you add the code for it?  I'm tied up with other things at the moment.

Reply:
Note that the query for GPU buffer, and the IPC handle creation need to be modified too.

Reply:
Will do.

Reply:
Emm, looking at the code again. `true_lb` is already added into mem handle. It should work correctly. Is there any test fails with this code?

```
    MPIDI_IPCI_mem_attr_t attr;
    MPIDI_IPCI_get_mem_attr((char *) buf + true_lb, &attr);

    if (rank != comm->rank && attr.ipc_type != MPIDI_IPCI_TYPE__NONE &&
```

Reply:
Yes, if you try Ken's PR, a bunch of tests are failing.